### DISCUSSION TITLE
Writing multiple recovery files to multiple media.

### DISCUSSION THREAD

**Discussion by Nik0s5** [Yutaka-Sawada/MultiPar#78] [Q&A]
Good afternoon! 
Thanks a lot for a great project. 
Is it possible to make sure that when creating several recovery files there is no message about the lack of space, but the process starts and prompts you to select a new location for the next file when the free space runs out. This would be handy when writing multiple recovery files to external media when the total size of the files exceeds the size of a single media.

**Comment by Yutaka-Sawada** [comment]
It will be difficult to implement. Basically, MultiPar (par2j) writes every recovery blocks on belonging recovery files on time. Only when there are enough memory to keep whole recovery blocks, it writes them on each recovery file one by one. In the latter case, it may be possible to check available space for each recovery file, before writing them. But, the mechanism will be complex. Also, you must have more free memory (RAM on PC) than the external media.

I don't know how is your usage. When you create 10 GB recovery files, you may want to save them on 5 DVD disks (2 GB * 5 disks = 10 GB). In this case, you need at least 10 GB free space on RAM to keep the recovery data temporary.

If you have PC with many RAM and don't want to save recovery files on HDD or SSD, a simple way is using RAM drive. Carete recovery files on a RAM drive temporary, and move them on external medias manually. Though the operation is manual, the behavior is same as what you want.
